platform/security/certificate/security-certificate-keystoreeditor/src/main/java/org/codice/ddf/security/certificate/keystore/editor/KeystoreEditor.java

```diff
@@ -266,6 +280,96 @@ public void addTrustedCertificate(String alias, String keyPassword, String store
     }
 
     @Override
+    public List<Map<String, Object>> addTrustedCertificateFromUrl(String url) {
+        SSLSocket socket = null;
+        String decodedUrl = null;
+        List<Map<String, Object>> resultList = new ArrayList<>();
+        try {
+            decodedUrl = new String(Base64.getDecoder()
+                    .decode(url), "UTF-8");
+            socket = getNonVerifyingSslSocket(decodedUrl);
+            socket.startHandshake();
+            X509Certificate[] peerCertificateChain = (X509Certificate[]) socket.getSession()
+                    .getPeerCertificates();
+            for (X509Certificate certificate : peerCertificateChain) {
+                try {
+                    X500Name x500name = new JcaX509CertificateHolder(certificate).getSubject();
+                    RDN cn = x500name.getRDNs(BCStyle.CN)[0];
+                    String cnStr = IETFUtils.valueToString(cn.getFirst()
+                            .getValue());
+                    trustStore.setCertificateEntry(cnStr, certificate);
+                    resultList.add(Collections.singletonMap("success", true));
+                } catch (CertificateEncodingException e) {
+                    resultList.add(Collections.singletonMap("success", false));
+                    LOGGER.error("Unable to store certificate: {}", certificate.toString(), e);
+                }
+            }
+            Path trustStoreFile = Paths.get(SecurityConstants.getTruststorePath());
+            if (!trustStoreFile.isAbsolute()) {
+                Path ddfHomePath = Paths.get(System.getProperty("ddf.home"));
+                trustStoreFile = Paths.get(ddfHomePath.toString(), trustStoreFile.toString());
+            }
+            String keyStorePassword = SecurityConstants.getTruststorePassword();
+            OutputStream fos = Files.newOutputStream(trustStoreFile);
+            trustStore.store(fos, keyStorePassword.toCharArray());
+        } catch (IOException | GeneralSecurityException e) {
+            LOGGER.error("Unable to add certificate(s) to trust store from URL: {}",
+                    (decodedUrl != null) ? decodedUrl : url, e);
+        } finally {
+            IOUtils.closeQuietly(socket);
+        }
+        return resultList;
+    }
+
+    @Override
+    public List<Map<String, Object>> showCertificateFromUrl(String url) {
```

Comment:
Might want to change to something like `certificateDetails(String url)`.  This function does not show anything and the FromUrl part is redundant with the parameter list.